web/satellite/src/components/dialogs/BucketDetailsDialog.vue

```diff
@@ -119,26 +120,43 @@ const showRegionTag = computed<boolean>(() => {
     return configStore.state.config.enableRegionTag;
 });
 
-function redirectToBucketsPage(): void {
-    router.push({
-        name: ROUTES.Buckets.name,
-        params: { id: projectsStore.state.selectedProject.urlId },
-    });
-}
-
-const bucket = computed((): Bucket => {
-    if (!projectsStore.state.selectedProject.id) return new Bucket();
+/**
+ * Fetch the bucket data if it's not avaliable
+ */
+async function loadBucketData() {
+    if (!projectsStore.state.selectedProject.id) {
+        bucket.value = new Bucket();
+        return;
+    }
 
     const data = bucketsStore.state.page.buckets.find(
         (bucket: Bucket) => bucket.name === props.bucketName,
     );
 
-    if (!data) {
-        redirectToBucketsPage();
-
-        return new Bucket();
+    if (data) {
+        bucket.value = data;
+    } else {
+        try {
```

Comment:
To make things a bit fancier, you could wrap this try-catch block with the existing loader handling composable `useLoading`.

It's used like this

```
const { isLoading, withLoading } = useLoading();
```

and our code would look like this

```
           withLoading(async () => {
                try {
                    bucket.value = await bucketsStore.getSingleBucket(selectedProjectID, props.bucketName);
                } catch (error) {
                    notify.notifyError(error, AnalyticsErrorEventSource.BUCKET_DETAILS_MODAL);
                    bucket.value = new Bucket();
                }
            });
```

this would allow us to set VCard's `loading` prop like this

```
<VCard :loading=isLoading>
...
</VCard>
```

By doing that, we'd add a small but stylish loading animation, which would be more noticeable in production.